torch/optim/asgd.py

```diff
@@ -376,6 +377,7 @@ def _multi_tensor_asgd(
             torch._foreach_add_(grouped_state_steps, 1)
 
         # intermediate = grad + param * lambd
+        intermediate: Union[Tuple[Tensor, ...], List[Tensor]]
```

Comment:
what’s the diff between defining this here vs the first time it’s instantiated?

Reply:
without this, mypy will raise an error `Incompatible types in assignment (expression has type "tuple[Tensor, ...]", variable has type "list[Tensor]")` in line 386, 392 and 411 because `intermediate` was assigned as list[Tensor] in line 384. I think it is more concise to assign a union type than add 3 type ignores.